### ISSUE TITLE
PHP Parse error: syntax error, unexpected '?'

### ISSUE THREAD

**Issue by sal0max** [sal0max/grav-plugin-shortcode-gallery-plusplus#27]
Received this one per mail:

> After update shortcode-gallery-plusplus 1.2.1 > 1.4 I got the following error:
> 
> ```php
> [Thu May 05 17:27:15.751955 2022] [php7:error] [pid 26888] [client 84.80.23.178:55825] PHP Parse error: syntax error, unexpected '?', expecting function (T_FUNCTION) or const (T_CONST) in /var/www/html2/grav/user/plugins/shortcode-gallery-plusplus/shortcode-gallery-plusplus.php on line 15
> ```
>
>
> My grav website and admin-page are not working any more. the only solution was to remove plugin by CLI and reinstal the old plugin (from backup)

---

The code in question here is [this line](https://github.com/sal0max/grav-plugin-shortcode-gallery-plusplus/pull/25/files#diff-d3482ae2c35f0f2a6eb804ce1924bc07bc67100cdef9627a96217f7dae14de70R15) from the merged pull request #25 from @Sailoriae:
```php
private ?Page $currentPage = null;
```

The error message indicates, that the `?`, which is a Nullable Type hint and which was [introduced in PHP 7.1](https://wiki.php.net/rfc/nullable_types), is wrong.

My guess: The email author uses PHP < 7.1, which causes this error. And that is odd. This plugin states, that it needs [at least Grav 1.6](https://github.com/sal0max/grav-plugin-shortcode-gallery-plusplus/blob/master/blueprints.yaml#L18), and Grav 1.6 needs [at least PHP 7.1.3](https://learn.getgrav.org/16/basics/requirements).

**Strong indication that PHP7 is used: `[php7:error]` in the error message. ;)**

So, tldr;: Author needs to provide more information. I'll keep this issue updated.

**Comment by Sailoriae** [comment]
Hello ! ^^
PHP 7.0 has reached EOL 3 years ago, and Grav requires PHP 7.3. It's maybe time to upgrade ! 😆
However, if you wish to support old versions, you can just remove the type of the `$currentPage` attribute.
```
private $currentPage = null;
```
Have a nice day ! ^^

**Comment by sal0max** [comment]
Thanks a lot for your help, @Sailoriae!

Done in 01db1836fb5a7bc929659b87320d75de0f46dde7
Implemented and released in version 1.4.1.
